client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -164,12 +165,14 @@ const siteSetupFlow: Flow = {
 					if ( ! siteSlug ) {
 						return;
 					}
+
+					const siteIntent = getIntent();
 					const siteId = site?.ID;
 					const pendingActions = [
-						setIntentOnSite( siteSlug, intent ),
+						setIntentOnSite( siteSlug, siteIntent ),
```

Comment:
why cant we use the regular `intent` here (and below)? Is there a reason we need to have both:

```
		const intent = useSelect( ( select ) => select( ONBOARD_STORE ).getIntent() );
		const { getIntent } = useSelect( ( select ) => select( ONBOARD_STORE ) );
```

Reply:
Im not able to get the correct value from using the existing `intent` variable. It appears an empty string.

Reply:
I was going to ask the same question. Any thoughts on why? I assume it may a JS timing issue related to how and when pendingAction is invoked? Like in one case, you're getting the value of intent NOW as you set up the function (essentially acts as a closure and stores the current value), and in the other, your fetching site intent later, when the method itself is fired. 

I will say it seems odd in this file to be fetching intent here twice via two different methods, in two different places.

Reply:
I think one other thing I'm puzzle by here is that before this PR, we WERE using the first/original intent value. Does that mean that code was not working as expected?

Reply:
I'm a bit concerned about whether this change (since it clearly does change the behavior and possible value of intent) will have unexpected impact on other existing flows beyond what we're wanting to do just for ours.

Reply:
Reviewed this @agrullon95 and just did some extensive testing of other scenarios and flows to confirm I'm not seeing any issues.